### PR TITLE
apply more encoding w/ tests

### PR DESCRIPTION
**Pull request by diosmosis**

### CODE REVIEW

**Comment by sgiehl** [comment]
@diosmosis not sure what problem you are trying to solve with the changes, but seems the tests are now failing on PHP 5.3.3 only. Not sure if that is relevant any longer. Maybe we could simply discontinue support for PHP 5.3 for this lib...

**Comment by diosmosis** [comment]
@sgiehl we recently merged a pr that fixed some output, in it it was seen that some fields will not escape for unsafe output allowing the INI to be injected (unless escaped by the caller). this was meant to fix that before releasing a new version of the component. The build fix should be simple.

**Comment by sgiehl** [comment]
ok. makes sense. Just out of curiosity: Is there a reason why we don't filter/escape `"` in values? If I see that correctly, that might also break the output, won't it?

**Comment by diosmosis** [comment]
:+1: didn't think of that, will look into that too

**Comment by diosmosis** [comment]
@sgiehl updated

File: tests/resources/Injection.clean.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+            'Under_Score_' => 6.4,
+            0 => 0,
+        ),
+        'injected-with_ab' => [
+            'test' => "value\n[newsection]\nc=d\n"
+        ],
+    ),
+    'a-poison_ ab' => [
+        'test' => "va\"lue\n[newsection]\nc=d\n"
+    ],
```
Comment (sgiehl) [inline tests/resources/Injection.clean.php:18]: If we want to keep compatibility for PHP 5.3.3 we need to use:
```suggestion
        'injected-with_ab' => array(
            'test' => "value\n[newsection]\nc=d\n"
        ),
    ),
    'a-poison_ ab' => array(
        'test' => "va\"lue\n[newsection]\nc=d\n"
    ),
```

File: tests/resources/Injection.php [hunk 2]
```diff
@@ -8,6 +8,12 @@
...
-        )
-    )
+        ),
+        "]\n[injected-with_] a=b\n" => [
+            'te"st' => "value\"\n[newsection]\nc=d\n",
+        ],
+    ),
+    "]\n[a-\"poison_] a=b\n" => [
+        'test' => "va\"lue\n[newsection]\nc=d\n",
+    ],
```
Comment (sgiehl) [inline tests/resources/Injection.php:18]:
```suggestion
        "]\n[injected-with_] a=b\n" => array(
            'te"st' => "value\"\n[newsection]\nc=d\n",
        ),
    ),
    "]\n[a-\"poison_] a=b\n" => array(
        'test' => "va\"lue\n[newsection]\nc=d\n",
    ),
```

**Review by sgiehl — APPROVED** [review]

**Pull request merged**